pkg/syncer/spec/spec_process.go

```diff
@@ -114,14 +115,30 @@ func (c *Controller) process(ctx context.Context, gvr schema.GroupVersionResourc
 	}
 	clusterName, name := clusters.SplitClusterAwareKey(clusterAwareName)
 
-	// to downstream
-	downstreamNamespace, err := shared.PhysicalClusterNamespaceName(shared.NamespaceLocator{
-		LogicalCluster: clusterName,
-		Namespace:      upstreamNamespace,
-	})
-	if err != nil {
-		klog.Errorf("Error hashing namespace %s|%s: %v", clusterName, upstreamNamespace, err)
-		return nil // ignore error, shouldn't happen
+	namespaceGvr := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}
+	desiredNSLocator := shared.NewNamespaceLocator(clusterName, c.workloadClusterLogicalClusterName, c.workloadClusterUID, c.workloadClusterName, upstreamNamespace)
+	downstreamNamespaces := c.downstreamInformers.ForResource(namespaceGvr).Informer().GetIndexer().List()
+
+	var downstreamNamespace string
+	for _, namespace := range downstreamNamespaces {
+		namespace := namespace.(*unstructured.Unstructured)
+		nsLocator, err := shared.LocatorFromAnnotations(namespace.GetAnnotations())
+		if err != nil {
+			continue
+		}
+		if reflect.DeepEqual(desiredNSLocator, *nsLocator) {
```

Comment:
why do this on the locator data type? Why not use the string, i.e. the annotation directly

Reply:
done